Dust fluxes output and add descriptions in the xml-namelist file

jmaerz opened the pull request:
Hi @TomasTorsvik and @JorgSchwinger , as discussed, this PR brings in dust output and some descriptions in the xml-namelist file. I tried to fill all iHAMOCC gaps to my best knowledge. I cleaned the way, M4AGO sinking velocity is handled in `mo_ocprod` (wasn't ideal before).

Together with this PR, I introduced the xmlchange switch `HAMOCC_SINKING_SCHEME` (and removed the compset HAMOCC_M4AGO switch) that allows to select among the four supported sinking schemes: [WLIN, AGG, M4AGO, CONST], the default remains WLIN.

Ideally, this PR also enters the v1.6.x release version. Sorry for this delayed PR for the release tag - it just adds some potentially valuable output (and provides a more descriptive iHAMOCC xml-namelist definition file).

jmaerz committed:
Introduce dust flux output and add descriptions to iHAMOCC namelist definitions; fix preformed Si units in output

jmaerz committed:
switch off WLIN and AGG, if M4AGO is switched on

jmaerz committed:
add some more decriptions

JorgSchwinger commented on hamocc/mo_bgcmean.F90:
```diff
@@ -859,6 +867,19 @@ subroutine alloc_mem_bgcmean(kpie,kpje,kpke)
+      jdustflx2000(n)=i_bsc_m2d*min(1,FLX_DUST2000(n))
+      if (FLX_DUST4000(n) > 0) i_bsc_m2d=i_bsc_m2d+1
+      jdustflx4000(n)=i_bsc_m2d*min(1,FLX_DUST4000(n))
+      if (FLX_CAL_BOT(n) > 0) i_bsc_m2d=i_bsc_m2d+1
```
This should be FLX_DUST_BOT(n)!

jmaerz replied:
Thanks for catching this!

JorgSchwinger commented on hamocc/mo_ncout_hamocc.F90:
```diff
@@ -585,6 +594,12 @@ subroutine ncwrt_bgc(iogrp)
+    call wrtsrf(jdustflx1000(iogrp),  FLX_DUST1000(iogrp),  rnacc*1e3/dtbgc,0.,cmpflg,'dustflx1000')
+    call wrtsrf(jdustflx2000(iogrp),  FLX_DUST2000(iogrp),  rnacc*1e3/dtbgc,0.,cmpflg,'dustflx2000')
+    call wrtsrf(jdustflx4000(iogrp),  FLX_DUST4000(iogrp),  rnacc*1e3/dtbgc,0.,cmpflg,'dustflx4000')
+    call wrtsrf(jdustflx_bot(iogrp),  FLX_DUST_BOT(iogrp),  rnacc*1e3/dtbgc,0.,cmpflg,'dustflx_bot')
```
Did you check that the 1e3 is correct here? Usually this converts from kmol -> mol, is it here from kg -> g?

jmaerz replied:
Yes, it's from kg -> g.

JorgSchwinger reviewed (approved):
Looks good to me

jmaerz committed:
fix

TomasTorsvik commented:
Looks fine for me, but I will hold off on the review until the PR is updated with namelist option.

jmaerz committed:
Introduce HAMOCC_SINKING_SCHEME xml-switch [WLIN,AGG,M4AGO,CONST] to enable selection of sinking scheme - default: WLIN

jmaerz commented:
Hi @JorgSchwinger and @TomasTorsvik , I tinkered around with the component settings and the xml-namelist and found a way to get to the wanted behavior by making changes to `builnml`, `config_component.xml` and `namelist_definition_blom.xml` - so no further fortran code changes were needed.

I introduced a new xmlchange switch named `HAMOCC_SINKING_SCHEME` (and removed the compset HAMOCC_M4AGO switch) that allows to select among the four supported schemes: `[WLIN,AGG,M4AGO,CONST]`, default is `WLIN`. Buildnml sets then the `use_XXX` to the respective values (very explicit, maybe better ways are available).

I hope that satisfies the needs that we have. Tomas an I, we discussed that we could introduce another small subroutine that allows to replace all the different places of `if...else if ...` statements in `mo_ocprod` for choosing the sinking velocities by a simple function/subroutine call. This could be done for the `master` afterward - just to make the whole sinking routine a bit cleaner. I feel that this PR is already big for playing it on the v1.6.xxx release version, which is why one could just do it for `master`, if wished for - it would ease the maintenance at least.

TomasTorsvik reviewed (approved):
@jmaerz - thanks! Looks good to me.

jmaerz committed:
fix namelist entry

JorgSchwinger commented:
Looks fine, thank you.
